cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java

```diff
@@ -95,14 +96,30 @@ public StatefulSetOperator(Vertx vertx, KubernetesClient client, long operationT
         Future<Void> f = Future.succeededFuture();
         for (int i = 0; i < replicas; i++) {
             String podName = name + "-" + i;
-            String pvcName = AbstractModel.getPersistentVolumeClaimName(name, i);
             Pod pod = podOperations.get(namespace, podName);
 
             if (pod != null) {
                 if (Annotations.booleanAnnotation(pod, ANNO_STRIMZI_IO_DELETE_POD_AND_PVC,
                         false, ANNO_OP_STRIMZI_IO_DELETE_POD_AND_PVC)) {
-                    f = f.compose(ignored -> deletePvc(ss, pvcName))
-                            .compose(ignored -> maybeRestartPod(ss, podName, p -> true));
+
+                    f = f.compose(ignored -> {
+                        Map<String, String> ssLabels = ss.getMetadata().getLabels();
+                        // get all the PVCs created for a SS from the claim template
+                        Labels pvcSelector =
+                                Labels.forCluster(ssLabels.get(Labels.STRIMZI_CLUSTER_LABEL))
+                                        .withKind(ssLabels.get(Labels.STRIMZI_KIND_LABEL))
+                                        .withName(name);
```

Comment:
When we're creating the PVC "by hand" (rather than using VCT) we can add a label for selecting the PVC just using selector, without needing this client-side filtering, is that the plan?

Reply:
What do you mean using selector?

Reply:
If we're not relying on VCT to create the PVCs, we can use a label for identifying all the PVC for a given pod. I thought you'd tried that at one point, but the VCT only allows fixed labels on the PVC. But maybe I misunderstood something.

Reply:
First, when you said "selector" I started to think what we have in `Deployment` for example for selecting the matching pods :-) so I asked. 
Yeah, we had this discussion but it turned out that using VCT you cannot have a "specific" label per PVC because it uses a template so the label will be the same for all the PVCs for different pods, which is not we want. I guess it could be done when moving to create the PVCs by hand.